packages/auth/migrations/20230925183132_add_finalization_reason_grants.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.schema
+    .hasColumn('grants', 'finalizationReason')
+    .then((exists) => {
+      if (!exists) {
+        return knex.schema.alterTable('grants', (table) => {
+          table.enum('finalizationReason', ['ISSUED', 'REVOKED', 'REJECTED'])
+        })
+      }
+    })
```

Comment:
Let me know if this makes sense:
```suggestion
    .then((exists) => {
      if (!exists) {
        return knex.schema.alterTable('grants', (table) => {
          table.enum('finalizationReason', ['ISSUED', 'REVOKED', 'REJECTED'])
        })
      }
      return knex.schema.alterTable('grants', (table) => {
        table.enum('finalizationReason', ['ISSUED', 'REVOKED', 'REJECTED']).alter()
      })
    })
```

If the column does not exist, create it. 
If the column already exists, alter it to use check constraints.

Reply:
Unfourtunately wasn't as easy as adding `alter()` to change from string to enum type. I've been looking into several ways to do so, I've come up with a way, just going to be testing it